packages/nimble-components/src/table/tests/table.pageobject.ts

```diff
@@ -216,15 +189,24 @@ export class TablePageObject<T extends TableRecord> {
         }
     }
 
-    private getCell(rowIndex: number, columnIndex: number): TableCell {
+    public clickRow(rowIndex: number): void {
+        const row = this.getRow(rowIndex);
+        row.click();
+    }
+
+    public getRow(rowIndex: number): TableRow {
```

Comment:
Does `getRow` need to be public? If it's used from a test, could we instead expose a higher level function that encapsulates what the test is trying to learn about the row? Just trying to follow the principle we've established in systemlink-lib-angular page objects of preferring not to expose entire element APIs to tests.

Reply:
I had made `getRow` public so I could test the `aria-selected` attribute for rows within the table's tests. However, when I thought about it more, I decided that it makes more sense for the row itself to own its own `aria-selected` attribute. Therefore, I made the following changes:
- Made `getRow` private in the page object
- Removed the code within the table that calculated the `aria-selected` value for each row
- Added the ability for a row to know whether or not it was selectable
- Made the `nimble-table-row` calculate its own `aria-selected` value based on its "selectable" and "selected" states
- Wrote `aria-selected` tests within the row tests rather than the table tests
- Wrote tests for the table that verify `selectable` and `selected` are set on rows correctly